packages/ui/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,114 @@
+import { h, Component } from "preact";
+import * as styles from "./popover.scss";
+
+const VERTICAL_POSITIONS = ["top", "bottom"];
+const HORIZONTAL_POSITIONS = ["left", "right"];
+const TRIGGER_MOD = ["click", "hover"];
+
+const defaultProps = {
+    verticalPosition: VERTICAL_POSITIONS[0],
+    horizontalPosition: HORIZONTAL_POSITIONS[0],
+    triggerMod: TRIGGER_MOD[0],
+    className: "popover",
+    open: false,
+    closeOnEsc: true
+};
+
+interface PopoverProps {
+    onClose?: () => void;
+    onOpen?: () => void;
+    verticalPosition: "top" | "bottom";
+    horizontalPosition: "right" | "left";
+    className: string;
+    open: boolean;
+    closeOnEsc: boolean;
+    triggerMod: "click" | "hover";
+    anchorEl: JSX.Element;
+    children: JSX.Element | JSX.Element[];
+}
+
+export class Popover extends Component<PopoverProps> {
+    static defaultProps = {
+        ...defaultProps
+    };
+    componentDidMount() {
+        if (this.props.closeOnEsc && this.props.onClose) {
+            document.addEventListener("keydown", this._handleEscButtonPressed);
+        }
+    }
+    componentWillUnmount() {
+        document.removeEventListener("keydown", this._handleEscButtonPressed);
+    }
+    private _handleEscButtonPressed = (e: KeyboardEvent) => {
+        const { onClose } = this.props;
+        if (e.keyCode === 27 && onClose) {
+            onClose();
+        }
+    };
+    private _getTrigger = () => {
```

Comment:
this method is not relevant once you will use constants

Reply:
done